packages/@aws-cdk/aws-codedeploy/lib/lambda/deployment-group.ts

```diff
@@ -0,0 +1,196 @@
+  readonly deploymentGroupName: string;
+  readonly deploymentGroupArn: string;
+
+  export(): LambdaDeploymentGroupImportProps;
+}
+
+export interface LambdaDeploymentGroupProps {
+  application: LambdaApplication;
+  deploymentGroupName?: string;
+  deploymentConfig: LambdaDeploymentConfig;
+
+  alarms?: cloudwatch.Alarm[];
+
+  serviceRole?: iam.Role;
+  alias: lambda.Alias;
+
+  preHook?: lambda.IFunction;
+  postHook?: lambda.IFunction;
+
+  /**
+   * Whether to continue a deployment even if fetching the alarm status from CloudWatch failed.
+   *
+   * @default false
+   */
+  ignorePollAlarmsFailure?: boolean;
+
+  /**
+   * The auto-rollback configuration for this Deployment Group.
+   */
+  autoRollback?: AutoRollbackConfig;
+}
+export class LambdaDeploymentGroup extends cdk.Construct implements ILambdaDeploymentGroup {
+  /**
+   * Import an Lambda Deployment Group defined either outside the CDK,
+   * or in a different CDK Stack and exported using the {@link #export} method.
+   *
+   * @param parent the parent Construct for this new Construct
+   * @param id the logical ID of this new Construct
+   * @param props the properties of the referenced Deployment Group
+   * @returns a Construct representing a reference to an existing Deployment Group
+   */
+  public static import(scope: cdk.Construct, id: string, props: LambdaDeploymentGroupImportProps): ILambdaDeploymentGroup {
+    return new ImportedLambdaDeploymentGroup(scope, id, props);
+  }
+
+  public readonly application: ILambdaApplication;
+  public readonly deploymentGroupName: string;
+  public readonly deploymentGroupArn: string;
+
+  constructor(scope: cdk.Construct, id: string, props: LambdaDeploymentGroupProps) {
+    super(scope, id);
+
+    let serviceRole: iam.Role | undefined = props.serviceRole;
+    if (serviceRole) {
+      if (serviceRole.assumeRolePolicy) {
+        serviceRole.assumeRolePolicy.addStatement(new iam.PolicyStatement()
+          .addAction('sts:AssumeRole')
+          .addServicePrincipal('codedeploy.amazonaws.com'));
+      }
+    } else {
+      serviceRole = new iam.Role(this, 'ServiceRole', {
+        assumedBy: new iam.ServicePrincipal('codedeploy.amazonaws.com')
+      });
+    }
+    serviceRole.addToPolicy(new iam.PolicyStatement()
+      .addResource('arn:aws:s3:::*/CodeDeploy/*')
+      .addActions('s3:GetObject', 's3:GetObjectVersion'));
+    serviceRole.addToPolicy(new iam.PolicyStatement()
+      .addResource('*')
+      .addCondition('StringEquals', {
+        's3:ExistingObjectTag/UseWithCodeDeploy': 'true'
+      })
+      .addActions('s3:GetObject', 's3:GetObjectVersion'));
+    serviceRole.addToPolicy(new iam.PolicyStatement()
+      .addResource(props.alias.functionArn)
+      .addActions('lambda:UpdateAlias', 'lambda:GetAlias'));
+    if (props.alarms) {
+      serviceRole.addToPolicy(new iam.PolicyStatement()
+        .addResources(...props.alarms.map(alarm => alarm.alarmArn))
+        .addAction('cloudwatch:DescribeAlarms'));
```

Comment:
Can you explain all of these grants? I feel like some comments would be very helpful here .

Reply:
It's a narrow re-implementation of `arn:aws:iam::aws:policy/service-role/AWSCodeDeployRoleForLambda`. I've added some extra comments.

I'm going to do some testing around the S3 permissions ... i'm not sure why code deploy would need them, and i'm especially concerned since it is a wide set of permissions.